src/agent/centralized_configuration/src/centralized_configuration.cpp

```diff
@@ -6,18 +6,23 @@
 
 namespace centralized_configuration
 {
-    boost::asio::awaitable<module_command::CommandExecutionResult>
-    CentralizedConfiguration::ExecuteCommand(const std::string command) // NOLINT(performance-unnecessary-value-param)
+    boost::asio::awaitable<module_command::CommandExecutionResult> CentralizedConfiguration::ExecuteCommand(
+        const std::string command,       // NOLINT(performance-unnecessary-value-param)
+        const nlohmann::json parameters) // NOLINT(performance-unnecessary-value-param)
     {
         try
         {
-            const auto commnandAsJson = nlohmann::json::parse(command);
 
-            if (commnandAsJson["command"] == "set-group")
+            if (command == "set-group")
             {
                 if (m_setGroupIdFunction && m_downloadGroupFilesFunction)
                 {
-                    const auto groupIds = commnandAsJson["groups"].get<std::vector<std::string>>();
+                    if (parameters.empty())
+                        co_return module_command::CommandExecutionResult {
+                            module_command::Status::FAILURE,
+                            "CentralizedConfiguration group set failed, no group list"};
+
+                    const auto groupIds = parameters[0].get<std::vector<std::string>>();
 
                     m_setGroupIdFunction(groupIds);
 
```

Comment:
Code after this line is duplicated, maybe we can move it after the `if/else` statement.